db/migrate/20160131233303_remove_project_hidden_state_from_todos.rb

```diff
@@ -0,0 +1,42 @@
+class RemoveProjectHiddenStateFromTodos < ActiveRecord::Migration
+  class Todo < ActiveRecord::Base
+    has_many :successor_dependencies, :foreign_key => 'successor_id', :class_name => 'Dependency'
+    belongs_to :project
+  end
+
+  class Project < ActiveRecord::Base
+  end
+
+  class Dependency < ActiveRecord::Base
+    belongs_to :predecessor, :foreign_key => 'predecessor_id', :class_name => 'Todo'
+  end
+
+  def self.up
+    @todos = Todo.all
```

Comment:
Using `all` is a bit too expensive. I'd rather see this using something like `find_each` so that we're not loading all the todos into memory of which there could be thousands or hundreds of thousands (for larger installations)